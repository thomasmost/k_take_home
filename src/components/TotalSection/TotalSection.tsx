import styled from '@emotion/styled';
import * as React from 'react';
import { toDollarsFromCents } from '../../services/currency.svc';

interface ITotalSectionProps {
  // placeholder, remove or replace with real props
  balanceCents: number;
  availableCents: number;
  lockedCents: number;
}

const PillLayout = styled.div`
  background-color: white;
  border: 1px solid lightgray;
  border-radius: 10px;
  display: flex;
  padding: 20px;
  justify-content: space-between;
`;

const SubTotals = styled.div`
  display: flex;
  > div:first-child {
    padding-right: 30px;
    border-right: 1px solid lightgray;
  }
  > div:last-child {
    padding-left: 30px;
  }
`;

const Label = styled.label`
  font-size: 11px;
  margin-bottom: 6px;
  display: block;
  font-weight: 500;
`;

const SubTotal = styled.div`
  font-size: 14px;
`;

const Total = styled.div`
  font-size: 20px;
`;

export const TotalSection: React.FC<ITotalSectionProps> = ({
  balanceCents,
  availableCents,
  lockedCents,
}) => {
  return (
    <PillLayout>
      <div>
        <Label>Total Balance</Label>
        <Total>${toDollarsFromCents(balanceCents)}</Total>
      </div>
        <SubTotals>
        <div>
          <Label>Total Available</Label>
          <SubTotal>${toDollarsFromCents(availableCents)}</SubTotal>
        </div>
        <div>
          <Label>Total Locked</Label>
          <SubTotal>${toDollarsFromCents(lockedCents)}</SubTotal>
        </div>
      </SubTotals>
    </PillLayout>
  );
};
