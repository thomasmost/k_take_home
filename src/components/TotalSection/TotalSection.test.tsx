import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { TotalSection } from './TotalSection';

test('TotalSection renders', () => {
  const component = renderer.create(<TotalSection availableCents={11000} lockedCents={9000} balanceCents={20000} />);
  expect(component).toMatchSnapshot();
});
