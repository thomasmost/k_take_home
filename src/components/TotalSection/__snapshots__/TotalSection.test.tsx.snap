// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalSection renders 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-u8h7cj{background-color:white;border:1px solid lightgray;border-radius:10px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:20px;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;}",
      }
    }
    data-emotion="css u8h7cj"
  />,
  <div
    className="css-u8h7cj"
  >
    <div>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-77dcaw{font-size:11px;margin-bottom:6px;display:block;font-weight:500;}",
          }
        }
        data-emotion="css 77dcaw"
      />
      <label
        className="css-77dcaw"
      >
        Total Balance
      </label>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-end85d{font-size:20px;}",
          }
        }
        data-emotion="css end85d"
      />
      <div
        className="css-end85d"
      >
        $
        200.00
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1b534kz{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.css-1b534kz >div:first-child{padding-right:30px;border-right:1px solid lightgray;}.css-1b534kz >div:last-child{padding-left:30px;}",
        }
      }
      data-emotion="css 1b534kz"
    />
    <div
      className="css-1b534kz"
    >
      <div>
        <label
          className="css-77dcaw"
        >
          Total Available
        </label>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-mr3j8u{font-size:14px;}",
            }
          }
          data-emotion="css mr3j8u"
        />
        <div
          className="css-mr3j8u"
        >
          $
          110.00
        </div>
      </div>
      <div>
        <label
          className="css-77dcaw"
        >
          Total Locked
        </label>
        <div
          className="css-mr3j8u"
        >
          $
          90.00
        </div>
      </div>
    </div>
  </div>,
]
`;
